### ISSUE TITLE
With `on_class` hook defined, other hooks are not getting fired till the end of class

### ISSUE THREAD

**Issue by RKushnir** [rubocop-hq/rubocop#6113]
When traversing an AST, if you define an `on_class` hook in your parser object, other hooks till the end of parsed class get skipped.

This could be the intended behavior, parsing the class as a whole entity when a class hook is detected. But it wasn't obvious to me, that's why I brought it up.

--------

## Expected behavior

All hooks are called.

## Actual behavior

Only `on_class` hook is called

## Steps to reproduce the problem

Having a file `lab_rat.rb`:
```rb
class AClass
  a_macro
  another_macro
end
```

Try these snippets:

```ruby
class ParserWorksFine
  include RuboCop::AST::Traversal

  def initialize
    walk(RuboCop::ProcessedSource.from_file('lab_rat.rb', 2.5).ast)
  end

  def on_const(node)
    puts 'on const'
  end

  def on_send(node)
    puts 'on send'
  end
end

ParserWorksFine.new

# on const
# on send
# on send
```

```ruby
class ParserFailsWithOnClass
  include RuboCop::AST::Traversal

  def initialize
    walk(RuboCop::ProcessedSource.from_file('lab_rat.rb', 2.5).ast)
  end

  def on_class(node)
    puts 'on class'
  end

  def on_const(node)
    puts 'on const'
  end

  def on_send(node)
    puts 'on send'
  end
end

ParserFailsWithOnClass.new
# on class
```

## RuboCop version

```
$ rubocop -V
0.58.1 (using Parser 2.5.1.2, running on ruby 2.3.1 x86_64-darwin16)
```

**Comment by pocke** [comment]
The `Traversal` module needs calling `super`. For example:

```ruby
require 'rubocop'
class ParserFailsWithOnClass
  include RuboCop::AST::Traversal

  def initialize
    walk(RuboCop::ProcessedSource.from_file('lab_rat.rb', 2.5).ast)
  end

  def on_class(node)
    puts 'on class'
    super
  end

  def on_const(node)
    puts 'on const'
    super
  end

  def on_send(node)
    puts 'on send'
    super
  end
end

ParserFailsWithOnClass.new

# on class
# on const
# on send
# on send
```

You can find usage of the module from `RuboCop::Cop::Commissioner` class.
https://github.com/rubocop-hq/rubocop/blob/dbe86eec52d9a44f4702e0dcd48079d2fd7c9c2e/lib/rubocop/cop/commissioner.rb#L23-L36

**Comment by RKushnir** [comment]
Thanks @pocke
I was looking at the commissioner class, but missed this detail.
